src/NuGet.Core/NuGet.Common/MsBuildStringUtility.cs

```diff
@@ -85,19 +85,32 @@ public static bool IsTrueOrEmpty(string? value)
         }
 
         /// <summary>
-        /// Splits and parses a ; or , delimited list of log codes.
-        /// Ignores codes that are unknown.
+        /// Parses the specified string as a comma or semicolon delimited list of NuGet log codes and ignores unknown codes.
         /// </summary>
-        public static IEnumerable<NuGetLogCode> GetNuGetLogCodes(string s)
+        /// <param name="s">A comma or semicolon delimited list of NuGet log codes.</param>
+        /// <returns>An <see cref="IList{T}" /> containing the <see cref="NuGetLogCode" /> values that were successfully parsed from the specified string.</returns>
+        public static IList<NuGetLogCode> GetNuGetLogCodes(string s)
```

Comment:
Switching this to return `IList<NuGetLogCode>` may not be a win here. Looks like a lot of callers just want to enumerate the results (there is at least one that needs a `List`). Usage of `yield return` is going to allocate a state machine and creating a `List` allocates the list object and backing array. Depending on the size, I wouldn't be surprised if `yield return` won out.

Reply:
Talked offline, the point of this PR is to address calls to `new LibraryDependency()` not allocating a list and instead getting the list at construction.  This change does that but doesn't address places where the result of GetLogCodes() is used as an `IEnumerable>